apps/system/js/tracking_notice.js

```diff
@@ -0,0 +1,156 @@
+    this.controller = controller || {};
+    this.options = {};
+    this._settings = navigator.mozSettings;
+    var req = this._settings.createLock().get(TRACKING_NOTICE_KEY);
+    req.onsuccess = () => {
+      var alreadyShown = req.result[TRACKING_NOTICE_KEY];
+      this._alreadyShown = alreadyShown;
+      if (!alreadyShown) {
+        this.renderDialog();
+      }
+    };
+
+    req.onerror = () => {
+      this._alreadyShown = false;
+      this.renderDialog();
+    };
+
+    this.publish('created');
+  };
+
+  TrackingNotice.prototype = Object.create(SystemDialog.prototype);
+
+  TrackingNotice.prototype.customID = 'tracking-notice';
+
+  TrackingNotice.prototype.DEBUG = false;
+
+  TrackingNotice.prototype.name = 'TrackingNotice';
+
+  TrackingNotice.prototype.containerElement =
+    document.getElementById('system-overlay');
+
+  TrackingNotice.prototype.renderDialog = function() {
+    LazyLoader.load('shared/elements/gaia_switch/script.js', () => {
+      this.render();
+    });
+  };
+
+  TrackingNotice.prototype.view = function view() {
+    return `<section id="tracking-notice" data-z-index-level="system-dialog"
+              class='hidden dialog'>
+              <div class="panel">
+                <header><h1 data-l10n-id="tracking-notice-header"></h1></header>
+                <p class="copy">
+                  <span id="tracking-notice-copy"
+                  data-l10n-id="tracking-notice-description"></span>
+                  <a id="tracking-notice-learn-more"
+                    data-l10n-id="tracking-notice-learn-more">
+                  </a>
+                </p>
+                <div class="setting">
+                  <label data-l10n-id="tracking-protection"
+                  for="tracking-protection-toggle"></label>
+                  <gaia-switch id="tracking-protection-toggle">
+                  </gaia-switch>
+                  <p data-l10n-id="tracking-protection-setting-description"
+                  class="subtitle"></p>
+                </div>
+              </div>
+              <div class="confirm" id="tracking-notice-confirm">
+                <p class="copy" data-l10n-id="tracking-notice-confirm"></p>
+              </div>
+            </section>`;
+  };
+
+
+  TrackingNotice.prototype._registerEvents = function() {
+    var self = this;
+    self.confirm.addEventListener('click', function onConfirm() {
+      self.confirm.removeEventListener('click', onConfirm);
+      self.confirmNotice();
+    });
+    window.addEventListener('appopened', this.boundShowIfNeeded);
+  };
+
+  TrackingNotice.prototype._fetchElements = function spl_initElements() {
+    this.element = document.querySelector('#' + this.instanceID);
+    this.confirm = this.element.querySelector('#tracking-notice-confirm');
+    this.setting = this.element.querySelector('gaia-switch');
+    this.learnMore = this.element.querySelector('#tracking-notice-learn-more');
+    var boundShowIfNeeded = this.showIfNeeded.bind(this);
```

Comment:
I *really* don't like having this bound function when the convention is just to use 
```
 element.addEventListener('_locationchange', this);
```
and `handleEvent`. I won't block the review on it, but I suggest this is a cleaner way to go.